homeassistant/components/decora_wifi/config_flow.py

```diff
@@ -0,0 +1,122 @@
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BASE_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+class DecoreWifiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Decora Wifi Integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_data: dict[str, str]) -> FlowResult:
+        """Handle importting decora wifi config from configuration.yaml."""
+
+        async_create_issue(
+            self.hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.11.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Decora Wifi",
+            },
+        )
+
+        if CONF_USERNAME not in import_data or CONF_PASSWORD not in import_data:
+            _LOGGER.error(
+                "Could not import config data from yaml. Required Fields not found "
+                "in decora_wifi config: %s, %s",
+                CONF_USERNAME,
+                CONF_PASSWORD,
+            )
+            # We don't have enough to auto-import, so skip to new setup
+            return await self.async_step_user(None)
+
+        return await self.async_step_user(import_data)
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle a flow initiated by the user."""
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            username = user_input["username"]
+            password = user_input["password"]
+
+            try:
+                await async_validate_input(self.hass, username, password)
+                unique_id = username.lower()
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception as exc:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception: %s", exc)
+                errors["base"] = "unknown"
+            else:
+                # No Errors
+                existing_entry = await self.async_set_unique_id(unique_id)
+                self._abort_if_unique_id_configured()
+
+                if existing_entry:
+                    self.hass.config_entries.async_update_entry(
+                        existing_entry, data=user_input
+                    )
+                    # Reload the config entry otherwise devices will remain unavailable
+                    self.hass.async_create_task(
+                        self.hass.config_entries.async_reload(existing_entry.entry_id)
+                    )
```

Comment:
Why do we do this?

Reply:
honestly no idea, boilerplate. Should the entire if block be deleted?

Reply:
deleted here
https://github.com/home-assistant/core/pull/104573/commits/112fa3be2302716ce0f632ba4ebb4db2aced3eb5